client/client.go

```diff
@@ -214,6 +202,32 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// WaitForInclusion blocks until the requested data has been verified with an inclusion proof.
+func (c *LogClient) WaitForInclusion(ctx context.Context, data []byte) error {
+	leaf := c.logVerifier.buildLeaf(data)
+
+	// Fetch the current Root so we can detect .
+	if err := c.UpdateRoot(ctx); err != nil {
+		return err
+	}
+	err := c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
+	for {
+		if s, ok := status.FromError(err); !ok || s.Code() != codes.NotFound {
+			return err // Success or unexpected error.
+		}
+		if err := ctx.Err(); err != nil {
+			return err // Timeout.
+		}
```

Comment:
I'm more used to a pattern like:
```go
	select {
	case <-ctx.Done():
		return ctx.Err()
	default:
	}
```
Do you know if there's an official preferred way to check for cancellation?